feat(ai-topics): add Topic model, CRUD, routes and seed; topic-based …

akintewe opened the pull request:
- Prisma: add Topic model (name, description, category, englishLevel, prompts)
- Service: TopicService CRUD + list with level/category filters
- Controller: TopicController endpoints (create, update, delete, list, get by level, get by id)
- Routes: /api/v1/topics (GET list with ?level&category, GET /:level, GET /id/:id, POST, PUT, DELETE)
- Validation: Joi schemas for create/update/read
- ChatService: basic topic-based prompt builder placeholder
- Seed: add sample topics
- Wire: register topics routes in api index

# Pull Request  - Close Issue

❗ **Pull Request Information**

Implements a topic-based AI chat prompt system leveraging existing question categories and English levels (A1–C2). Adds a `Topic` model, CRUD APIs, validation, and a basic chat prompt builder that generates contextual prompts from selected topics. Includes seed data for immediate testing.

## 🗒️ Summary of Changes

- **Database (Prisma)**: Added `Topic` model with fields `id`, `name`, `description`, `category`, `englishLevel`, `prompts`
- **Service**: `TopicService` with CRUD and listing filtered by `level` and `category`
- **Controller**: `TopicController` with create, update, delete, list, get by level, get by id
- **Routes**: Registered under `/api/v1/topics`
  - `GET /api/v1/topics` (filters: `?level=&category=`)
  - `GET /api/v1/topics/:level`
  - `GET /api/v1/topics/id/:id`
  - `POST /api/v1/topics`, `PUT /api/v1/topics/:id`, `DELETE /api/v1/topics/:id`
- **Validation**: Joi schemas for create/update/read endpoints
- **Chat Service**: Basic topic-based prompt builder (`ChatService.buildPromptFromTopic`)
- **Seed Data**: Added sample topics (A1 “Food & Restaurants”, B1 “Travel”)
- **Wiring**: Registered topic routes in `routes/api/index.ts`

## ⚒️ Testing

- Run migrations and seed:
  - `npx prisma migrate dev --schema prisma/schema.prisma --name add_topic_model`
  - `npx prisma db seed`
- Verify endpoints (requires Authorization: Bearer <token>):
  - `GET /api/v1/topics?level=A1&category=daily_life` → 200 with topics
  - `GET /api/v1/topics/A1` → 200 with A1 topics
  - `GET /api/v1/topics/id/:id` → 200 with topic details
  - Create/Update/Delete protected endpoints return expected responses

## 📂 Related Issue

This pull request will **close #48** upon merging.

---



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Introduced “Topics” with levels and categories, including create, update, delete, list, and get-by-id/level endpoints with validation and auth.
  - Chat now generates prompts based on selected topic and user context.

- Enhancements
  - Added environment-driven, idempotent seeding with sample topics and improved error handling.
  - Improved database reliability with dynamic SSL support and non-fatal pool error handling.
  - Optimized topic filtering by level and category.

- Chores
  - Removed bundled Postgres service from docker-compose.
  - Ignored additional local files in version control.
  - Added development type dependency.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

akintewe committed:
feat(ai-topics): add Topic model, CRUD, routes and seed; topic-based chat prompt builder\n\n- Prisma: add Topic model (name, description, category, englishLevel, prompts)\n- Service: TopicService CRUD + list with level/category filters\n- Controller: TopicController endpoints (create, update, delete, list, get by level, get by id)\n- Routes: /api/v1/topics (GET list with ?level&category, GET /:level, GET /id/:id, POST, PUT, DELETE)\n- Validation: Joi schemas for create/update/read\n- ChatService: basic topic-based prompt builder placeholder\n- Seed: add sample topics\n- Wire: register topics routes in api index

akintewe committed:
refactor(topic): align schema with enums/arrays, idempotent seed, better errors; routes and controller tweaks; remove unused types

akintewe committed:
chore: address CodeRabbit suggestions (scripts path, SQL quoting, db SSL cfg); typing improvements for Topic service

akintewe committed:
chore(demo): improve readiness checks, configurable health URL, hardened login; escape ; fix SQL quoting

Gerson2102 commented:
Hello @akintewe. Thanks for your contribution. Can you fix the format on your PR please, and also provide a clear step by step process on how to run all your changes to check if everything is working fine. Thanks again.

akintewe committed:
fix: remove as any casts from topic service; address CodeRabbit suggestions

akintewe committed:
Delete backend/TESTING_INSTRUCTIONS.md

akintewe committed:
fix: update Prisma schema with proper enum mappings and table names

akintewe committed:
docs: add comprehensive testing instructions for Supabase setup and troubleshooting

akintewe committed:
fix: address all CodeRabbit suggestions - improve type safety and error handling

akintewe committed:
fix: resolve all remaining CodeRabbit issues - schema, types, and docs

akintewe committed:
chore: remove TESTING_INSTRUCTIONS.md from git tracking and add to .gitignore

akintewe committed:
chore: remove .env.backup from git tracking and add to .gitignore

akintewe committed:
feat: implement all CodeRabbit improvements - add Category enum, improve type safety, add indexes, normalize inputs, and move testing docs to proper location

akintewe committed:
fix: remove documentation and implement proper Category enum validation in topic validators

akintewe committed:
fix: address remaining CodeRabbit suggestions - remove unused @types/urijs dependency - add cd to backend root in demo script - fix demo script routes and category values - add TESTING_INSTRUCTIONS.md to .gitignore

Gerson2102 commented:
I'm getting the following error while trying to run the npx prisma stuff:

```console

Drift detected: Your database schema is not in sync with your migration history.

The following is a summary of the differences between the expected database schema given your migrations files, and the actual schema of the database.

It should be understood as the set of changes to get from the expected schema to the actual schema.

[*] Changed the `QuestionType` enum
  [+] Added variant `idiom_challenge`

[*] Changed the `User` table
  [+] Added column `lastActivityAt`
  [+] Added column `lastLoginAt`
  [+] Added column `lastReminderSent`

```

I dont know how to fix it, the solution is to drop the database but I dont know if I should do that. I'm talking with my team.

Gerson2102 commented:
I dont know if I had to run something after the reset.

akintewe commented:
> I dont know if I had to run something after the reset.

@Gerson2102 This is good , now you're seeing the correct schema drift  which has our new Topic system and improvements. You should do this

**Step 1: Reset the database schema**
```bash
npx prisma migrate reset
```
This will automatically:
- Drop the current schema
- Apply our migrations  
- Run the seed script

**Step 2: Verify the schema is in sync**
```bash
npx prisma db push
```

**Step 3: Start the server and test**
```bash
npm run start:prod
```

**Step 4: Test our implementation**
```bash
bash scripts/demo_topics.sh
```

The `migrate reset` command will handle everything automatically - you don't need to run anything else after it. It will create a fresh database with our Topic system and seed it with sample data.

akintewe commented:
> > > I dont know if I had to run something after the reset.
> > 
> > 
> > @Gerson2102 This is good , now you're seeing the correct schema drift which has our new Topic system and improvements. You should do this
> > **Step 1: Reset the database schema**
> > ```shell
> > npx prisma migrate reset
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This will automatically:
> > 
> > * Drop the current schema
> > * Apply our migrations
> > * Run the seed script
> > 
> > **Step 2: Verify the schema is in sync**
> > ```shell
> > npx prisma db push
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > **Step 3: Start the server and test**
> > ```shell
> > npm run start:prod
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > **Step 4: Test our implementation**
> > ```shell
> > bash scripts/demo_topics.sh
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The `migrate reset` command will handle everything automatically - you don't need to run anything else after it. It will create a fresh database with our Topic system and seed it with sample data.
> 
> I ran all of these and it went good except for the nom run start:prod, Im gonna use npm run dev.

@Gerson2102  yes you are right

What `migrate reset` already did for you was:
 Applied all our migrations
Created the Topic table and enums  
Ran the seed script automatically
Set up all indexes and constraints

What you need to do now is to : 

**Step 1: Generate Prisma client (if you havent done it before)**
```bash
npx prisma generate
```

**Step 2: Start the server**
```bash
npm run dev
```


**Step 3: Test our Topic system**
```bash
bash scripts/demo_topics.sh
```

Gerson2102 commented:
Great, now what is throwing me an error is in the login part of your script. I did not understand well how to do that in the past comments, can you clarify that?

In my env what I have is my gmail with its password, do I have to change it?

Gerson2102 commented:
Well in the script it already has the data and im not changing anything and is not working, even though about this you recommend me to do the first option?

akintewe committed:
fix: make demo script Supabase-compatible and fix API routes

akintewe committed:
fix: reinstall @types/urijs as it's required by @stellar/stellar-sdk

akintewe commented:
> Well in the script it already has the data and im not changing anything and is not working, even though about this you recommend me to do the first option?

@Gerson2102  I've fixed the demo script to work with Supabase. The script now:

Automatically detects if you're using Docker or Supabase
Provides Supabase instructions when Docker isn't available
Uses correct API routes for all operations
No manual editing needed everything is automatic

What to do now:
you can pull the changes

 Start your server:
   ```bash
   npm run dev
   ```

Run the demo script:
   ```bash
   bash scripts/demo_topics.sh
   ```

akintewe commented:
> I will finish reviewing later, thanks for all the guidance so far.

no problem

Gerson2102 commented:
> > Well in the script it already has the data and im not changing anything and is not working, even though about this you recommend me to do the first option?
> 
> @Gerson2102 I've fixed the demo script to work with Supabase. The script now:
> 
> Automatically detects if you're using Docker or Supabase Provides Supabase instructions when Docker isn't available Uses correct API routes for all operations No manual editing needed everything is automatic
> 
> What to do now: you can pull the changes
> 
> Start your server:
> 
> ```shell
> npm run dev
> ```
> 
> Run the demo script:
> 
> ```shell
> bash scripts/demo_topics.sh
> ```

Hello mate @akintewe 

I tried to run the script and I'm getting this error:

<img width="621" height="177" alt="image" src="https://github.com/user-attachments/assets/6c583364-7a71-4aa2-969a-3aa7a93977e3" />

akintewe committed:
fix: improve demo script with better error handling and troubleshooting

- Add comprehensive error handling for login failures
- Provide clear troubleshooting steps and solution options
- Create setup_demo.sh script for easy environment setup
- Add better visual feedback and progress indicators
- Handle both seeded user and custom credentials scenarios
- Improve error messages and user guidance

akintewe committed:
fix: resolve macOS compatibility issue in demo script

- Replace 'head -n -1' with 'sed $d' for macOS compatibility
- Demo script now works perfectly on both Linux and macOS
- All 8 steps complete successfully including login, CRUD operations, and chat prompt generation

akintewe commented:
> > > Well in the script it already has the data and im not changing anything and is not working, even though about this you recommend me to do the first option?
> > 
> > 
> > @Gerson2102 I've fixed the demo script to work with Supabase. The script now:
> > Automatically detects if you're using Docker or Supabase Provides Supabase instructions when Docker isn't available Uses correct API routes for all operations No manual editing needed everything is automatic
> > What to do now: you can pull the changes
> > Start your server:
> > ```shell
> > npm run dev
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Run the demo script:
> > ```shell
> > bash scripts/demo_topics.sh
> > ```
> 
> Hello mate @akintewe
> 
> I tried to run the script and I'm getting this error:
> 
> <img alt="image" width="621" height="177" src="https://private-user-images.githubusercontent.com/71728860/478789246-6c583364-7a71-4aa2-969a-3aa7a93977e3.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU0NTUzMjAsIm5iZiI6MTc1NTQ1NTAyMCwicGF0aCI6Ii83MTcyODg2MC80Nzg3ODkyNDYtNmM1ODMzNjQtN2E3MS00YWEyLTk2OWEtM2FhN2E5Mzk3N2UzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODE3VDE4MjM0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIwZmU0NmExMWMwOTAyZTY2MDU3NzYyZWRhMzI5NTVhOTM4ZDcxODEzZDdkNzRiNDU2ZTAyOGU0OWNlNThiZTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.PRYA55Wb-M1vjI2J29uCyUIPBtYAd5YOVyR4jJXy9bQ">
> And this option is not working too:
> 
> <img alt="image" width="616" height="262" src="https://private-user-images.githubusercontent.com/71728860/478789878-2e8b7491-5cab-4d4d-a435-ac9b99869a37.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU0NTUzMjAsIm5iZiI6MTc1NTQ1NTAyMCwicGF0aCI6Ii83MTcyODg2MC80Nzg3ODk4NzgtMmU4Yjc0OTEtNWNhYi00ZDRkLWE0MzUtYWM5Yjk5ODY5YTM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODE3VDE4MjM0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk1NzJjNTQ3NjgzOGVkNTA0YTQ2Yzc1NWJmNzkyODdkNzRjNWVkMjI1YzI5YjMxZDU1OGRjNjdmMmEyN2I5MTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.42i3am8kYU9inPejws3wzIouIIK90x4X2Eai8iT2vBM">

@Gerson2102 Hello  👋
I've fixed the login issue you encountered! The issue is that you're trying to use custom credentials (test@example.com / testpassword123) but those don't exist in the database.

I've created a comprehensive setup script that will fix everything automatically. Copy and paste these commands exactly:
Step 1: Setup Everything (IMPORTANT - Do This First!)

cd backend
bash scripts/setup_demo.sh

Step 2: Run the Demo

bash scripts/demo_topics.sh

Important: DON'T Change the Credentials!
DO NOT set custom environment variables like:

# ❌ DON'T DO THIS
export LOGIN_EMAIL="test@example.com"
export LOGIN_PASSWORD="testpassword123"

The scripts are configured to use the correct test user:
Email: customer@aurora.com
Password: password123!


After setup, you should see:

�� Setup Complete! 🎉
�� Test user created: customer@aurora.com
�� Server running on: http://localhost:8000

akintewe committed:
fix: address CodeRabbit suggestions for demo script

- Remove -f flag from curl to allow proper HTTP status handling (400/401)
- Fix PUT/DELETE routes to use correct endpoint format (/topics/:id)
- Use unique topic name for testing to avoid constraint violations
- All 8 demo steps now complete successfully including full CRUD operations

Gerson2102 commented:
Hello @akintewe.

I'm getting this error when I try to run the demo:

```

[2/8] Logging in to get JWT token...
🔑 Attempting login with: customer@aurora.com
❌ Login failed (HTTP 400)
📋 Response saved to /tmp/login_response.json

🔧 TROUBLESHOOTING:
   The user 'customer@aurora.com' doesn't exist or password is wrong.

💡 SOLUTION OPTIONS:
   Option 1: Create the test user by running:
     npx prisma db seed

   Option 2: Use your own credentials:
     export LOGIN_EMAIL='your-email@example.com'
     export LOGIN_PASSWORD='your-password'
     bash scripts/demo_topics.sh

   Option 3: Check if database is seeded:
     npx prisma studio

📊 Current login response:
{
  "success": false,
  "message": "Invalid credentials"
}

```

Gerson2102 commented:
I already ran the setup_demo script and then the demo script and still getting the same error. And the database is connected. 

Also I ran this:

npx prisma db seed
Environment variables loaded from .env

And still getting the same error.

akintewe committed:
fix: improve Supabase connection handling and prevent server crashes

- Increase connection timeout from 2s to 10s
- Reduce connection pool size from 20 to 5 for Supabase compatibility
- Handle database pool errors gracefully without crashing server
- Fix SSL configuration for Supabase connection stability

akintewe commented:
@Gerson2102 Here's exactly what to do to test the fixed system:
cd backend
git pull origin feat/topic-chat-system-48

then you do this

bash scripts/setup_demo.sh

bash scripts/demo_topics.sh

let me know if it works now, i have made a fix  

Login will work with: customer@aurora.com / password123!

Gerson2102 commented:
When I'm running the script and is trying to login I'm getting the error, is not working.

Maybe because I dont have that user: [customer@aurora.com](mailto:customer@aurora.com) / password123! registered? Do I have to registered first? I tried that and same result, same error.

akintewe commented:
> When I'm running the script and is trying to login I'm getting the error, is not working.
> 
> Maybe because I dont have that user: [customer@aurora.com](mailto:customer@aurora.com) / password123! registered? Do I have to registered first? I tried that and same result, same error.

Have you seeded the data

Gerson2102 commented:
> also
> 
> @Gerson2102 Since you're already running the setup script first, let's debug this step by step:
> 
> 1. Does npx prisma db push --accept-data-loss work without errors?
> 2. Can you open Prisma Studio and check if the user actually exists?
> 
> npx prisma studio
> 
> 3. What error message are you getting exactly when trying to login?

1. Yes
2. I can open it but the user is not there and I dont know why. 
3. [2/8] Logging in to get JWT token...
🔑 Attempting login with: customer@aurora.com
❌ Login failed (HTTP 400)
📋 Response saved to /tmp/login_response.json

🔧 TROUBLESHOOTING:
   The user 'customer@aurora.com' doesn't exist or password is wrong.

💡 SOLUTION OPTIONS:
   Option 1: Create the test user by running:
     npx prisma db seed

   Option 2: Use your own credentials:
     export LOGIN_EMAIL='your-email@example.com'
     export LOGIN_PASSWORD='your-password'
     bash scripts/demo_topics.sh

   Option 3: Check if database is seeded:
     npx prisma studio

📊 Current login response:
{
  "success": false,
  "message": "Invalid credentials"
}

akintewe commented:
> > also
> > @Gerson2102 Since you're already running the setup script first, let's debug this step by step:
> > 
> > 1. Does npx prisma db push --accept-data-loss work without errors?
> > 2. Can you open Prisma Studio and check if the user actually exists?
> > 
> > npx prisma studio
> > 
> > 3. What error message are you getting exactly when trying to login?
> 
> 1. Yes
> 2. I can open it but the user is not there and I dont know why.
> 3. [2/8] Logging in to get JWT token...
>    🔑 Attempting login with: [customer@aurora.com](mailto:customer@aurora.com)
>    ❌ Login failed (HTTP 400)
>    📋 Response saved to /tmp/login_response.json
> 
> 🔧 TROUBLESHOOTING: The user '[customer@aurora.com](mailto:customer@aurora.com)' doesn't exist or password is wrong.
> 
> 💡 SOLUTION OPTIONS: Option 1: Create the test user by running: npx prisma db seed
> 
> Option 2: Use your own credentials: export LOGIN_EMAIL='[your-email@example.com](mailto:your-email@example.com)' export LOGIN_PASSWORD='your-password' bash scripts/demo_topics.sh
> 
> Option 3: Check if database is seeded: npx prisma studio
> 
> 📊 Current login response: { "success": false, "message": "Invalid credentials" }

ok i see , give me a minute

akintewe committed:
fix: make seed script idempotent to handle existing users gracefully

- Changed user creation from create() to upsert() to handle existing users
- Changed wallet creation from create() to upsert() to handle existing wallets
- Added better error handling and success messages
- Fixed issue where seeding would fail if user already existed
- Now seeding works reliably whether database is empty or has existing data

akintewe commented:
ok i just tested and its working, i made alittle fix, try this


cd backend
git pull origin feat/topic-chat-system-48
npx ts-node prisma/seed.ts
npm run start:prod
bash scripts/demo_topics.sh

akintewe committed:
feat: implement CodeRabbit suggestions for production safety and credential management

akintewe committed:
fix: make wallet address configurable to avoid unique constraint violations

akintewe committed:
fix: improve Prisma cleanup and exit handling per CodeRabbit suggestion

- Use finally block to ensure Prisma client always disconnects
- Replace process.exit(1) with process.exitCode = 1 for cleaner tooling
- Prevents hanging seed processes and improves CI/CD compatibility

Gerson2102 commented:
Great. Looks like everything is working now:

```console

bash scripts/demo_topics.sh
🚀 AURORA Topic System Demo Script
==================================
[1/8] Ensuring API server is running...
✅ Server is up.
[2/8] Logging in to get JWT token...
🔑 Attempting login with: customer@aurora.com
✅ Token acquired successfully!
[3/8] List all topics...
📋 Fetching all topics...
[4/8] Filter topics by level=A1 & category=DAILY_LIFE...
🔍 Filtering topics by A1 level and DAILY_LIFE category...
[5/8] Get topics by level (A1)...
📚 Getting topics for A1 level...
[6/8] Get topic by id (4a5c184e-07db-48ca-9635-50f73dd54d72)...
🔍 Fetching topic details for ID: 4a5c184e-07db-48ca-9635-50f73dd54d72
[7/8] Create, Update and Delete a topic...
➕ Creating new topic...
📋 Create response saved to /tmp/topic_create.json
✏️ Updating topic 1051251c-1531-424a-8ca8-e2704e04a847...
🗑️ Deleting topic 1051251c-1531-424a-8ca8-e2704e04a847...
[8/8] Database check...
🐳 Docker not available or container not running.
📊 For Supabase users: Check your database dashboard at https://supabase.com/dashboard
   Navigate to Table Editor > Topic table to see the data
   Or use the API response above to verify data creation
[BONUS] Generate a chat prompt from first topic...
🤖 Generating AI chat prompt...
Topic: Food & Restaurants
Prompt:
 You are an English practice partner. Topic: Food & Restaurants (Level: A1).
Have a short conversation with the user on this topic. Keep responses appropriate for the level.
Prompt: You are at a restaurant. Order your favorite meal.

🎉 Demo complete! 🎉
=====================
📁 Artifacts saved under /tmp:
   • login_response.json - Login attempt details
   • topics_all.json - All topics response
   • topics_filter.json - Filtered topics response
   • topics_level.json - Topics by level response
   • topic_by_id.json - Individual topic response
   • topic_create.json - Topic creation response
   • topic_update.json - Topic update response
   • topic_delete.json - Topic deletion response

📊 Database data: Check Supabase dashboard or use the API responses above.
🔑 Login credentials used: customer@aurora.com

✅ If you see all steps completed successfully, the topic system is working!

```

Good job and thanks a lot for all the guidance. Are those additional files like the scripts and that seed.ts file needed for this PR to work well? Or at least needed in the future to this functionality to work in the project?

I'm asking that because maybe those files can be deleted and not included in main. Please let me know to merge this PR.

Thanks again.

akintewe committed:
chore: remove demo scripts and docker files for cleaner PR

- Remove demo_topics.sh and setup_demo.sh (development/testing only)
- Remove docker-compose.yml (not needed for Supabase deployment)
- Keep core topic system functionality intact

akintewe commented:
@Gerson2102   The PR is now clean and ready!
What I removed (not needed in production):
Demo scripts (just for testing)
Docker files (not needed for Supabase)
What stays (essential functionality):
Complete topic system
Enhanced question system
Database schema and migrations
All API endpoints and validation
